### ISSUE TITLE
wrong directory is mounted in container workspace

### ISSUE THREAD

**Issue by hogegex** [microsoft/vscode-remote-release#5902]
- VSCode Version: 1.62.2 (user setup)
- Local OS Version:  Windows 10 20H2 (19042.1348)
- Remote OS Version: Unspecified
- Remote Extension/Connection Type: Docker (Remote-Containers v0.205.2)
- Logs: N/A

Steps to Reproduce:

I have one repository for a product.
And stored the code of a development assists tool wrote by Python in a sub directory of that repository.
This tool is developed using Remote-Containers.

```text
<repo_root>
+ .git // repository under version control
-- xx.sln // main product solution
+ SomeCSProject
  - * .cs, etc ... // main project codes
+ Tools
  + MyDevTool
    -- foo.py // code of development assists tool
    -- Dockerfile // devenv definition for tool
    + .devcontainer
      -- devcontainer.json // "dockerFile": "../Dockerfile"
```

When I open Tools/MyDevTool with Remote-Containers, a strange phenomenon occurs.
The Docker image used to launch the container is Tools/MyDevTool/Dockerfile as expected.
However, <repo_root> is mounted under /workspace of the started Docker container.
What I expected was that <repo_root>/Tools/MyDevTool would be mounted.

Removing or renaming <repo_root>/.git directory will mount <repo_root>/Tools/ MyDevTool as expected.
In other words, if the parent directory is version controled by git, it seems that a different directory than the tried to open will be opened.

--------

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by chrmarti** [comment]
We mount the folder containing the `.git` folder so Git can work inside the container. Is this causing any issues?

**Comment by hogegex** [comment]
[reproduce.zip](https://github.com/microsoft/vscode-remote-release/files/7600271/reproduce.zip)

Attachments are minimum reproducible example.

# Steps:

1. Run VSCode and Use `Open Folder in Container` command to open `reproduce\ChildDir\GrandChildDir`.

2. Check container's /workspaces/ directory.

   ```
   # ls /workspaces
   GrandChildDir
   
   # pwd
   /workspaces/GrandChildDir
   ```

   => It looks okay.

3. Close VSCode and return to container host.

4. Remove docker containers launched by vscode if exists.

   (Issues may not be reproduced if containers remains)

5. Execute `git init` command in `reproduce/` directory.

6. Do step 1 and 2 again.

   ```
   # ls /workspaces
   reproduce
   
   # pwd
   /workspaces/reproduce/ChildDir/GrandChildDir
   ```

   => "reproduce" directory is mounted instead of the specified "GrandChildDir" directory.

Is this behavior by design?

**Comment by chrmarti** [comment]
It is by design. The advantage of doing this is that you can run `git` (and the Git integration in VS Code) inside the container. That wouldn't work if the folder containing the `.git` folder wasn't mounted (the 'reproduce' folder in your case).
